src/components/CourseFilter/CourseFilter.tsx

```diff
@@ -0,0 +1,191 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+import {
+  Box,
+  Autocomplete,
+  TextField,
+  Button,
+  Typography,
+  FormControl,
+  ButtonGroup,
+} from '@mui/material';
+import { useTheme } from '@mui/material/styles';
+import { Key, useCallback } from 'react';
+import { CourseWithTagsAndStudentCount } from '@/lib/prisma/courses';
+import { usePathname, useSearchParams, useRouter } from 'next/navigation';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+type CourseFilterProps = {
+  initialCourses: CourseWithTagsAndStudentCount[];
+  initialTags: any;
+};
+
+export default function CourseFilter({
+  initialCourses,
+  initialTags,
+}: CourseFilterProps) {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  const { palette } = useTheme();
+
+  const [startDate, setStartDate] = useState();
+  const [endDate, setEndDate] = useState();
+  const [courseName, setCourseName] = useState('');
```

Comment:
```suggestion
  const [courseName, setCourseName] = useState('null');
```

Reply:
The MUI Autocomplete component in its combo box variant generates a lot of warnings in the console if the courseName variable is set to an empty string. Use null to get rid of the warnings (same thing on line 63).